invokeai/frontend/web/src/features/modelManagerV2/subpanels/ModelPanel/RelatedModels.tsx

```diff
@@ -0,0 +1,300 @@
+        </Flex>
+        {errors.map((error) => (
+          <FormErrorMessage key={error}>{error}</FormErrorMessage>
+        ))}
+      </FormControl>
+      <Box>
+        <Flex gap="2" flexWrap="wrap">
+          {
+            // Render the related model tags as styled components.
+            //
+            // Models are grouped visually by type, sorted with 'main' and 'lora' types at the front.
+            // A vertical Divider is inserted when the type changes between adjacent models.
+            // Tags include:
+            //   - Colored background based on model type (via getModelTagColor)
+            //   - Tooltip showing "<ModelType>: <ModelName>"
+            //   - Ellipsis-truncated tag name for compact layout
+            //   - A close button to remove the relationship
+            [...relatedModels]
+              .sort((aKey, bKey) => {
+                const a = modelConfigs?.entities[aKey];
+                const b = modelConfigs?.entities[bKey];
+                if (!a || !b) {
+                  return 0;
+                }
+
+                // Floats Mains and LoRAs to the front
+                const aPriority = MODEL_TYPE_PRIORITY.indexOf(a.type);
+                const bPriority = MODEL_TYPE_PRIORITY.indexOf(b.type);
+
+                const aScore = aPriority === -1 ? 99 : aPriority;
+                const bScore = bPriority === -1 ? 99 : bPriority;
+
+                return aScore - bScore || a.type.localeCompare(b.type) || a.name.localeCompare(b.name);
+              })
+              .reduce<JSX.Element[]>((acc, id, index, arr) => {
+                const model = modelConfigs?.entities[id];
+                if (!model) {
+                  return acc;
+                }
+
+                const modelName = model.name ?? id;
+                const modelType = model.type ?? 'unknown';
+                const modelTypeLabel = modelType.replace(/_/g, ' ').replace(/\b\w/g, (c) => c.toUpperCase());
+
+                //  Create a divider if the previous model is of a different type. Just a small dash of visual flair.
+                const prevId = index > 0 ? arr[index - 1] : undefined;
+                const prevModel = prevId ? modelConfigs?.entities[prevId] : null;
+                const needsDivider = prevModel && prevModel.type !== model.type;
+
+                if (needsDivider) {
+                  acc.push(<Divider orientation="vertical" key={`divider-${id}`} opacity={0.3} />);
+                }
+
+                acc.push(
+                  <Tag key={id} py={2} px={4} bg={`${getModelTagColor(model.type)}.700`}>
+                    <Tooltip label={`${modelTypeLabel}: ${modelName}`} hasArrow>
+                      <TagLabel
+                        style={{
+                          maxWidth: '50px',
+                          overflow: 'hidden',
+                          textOverflow: 'ellipsis',
+                          whiteSpace: 'nowrap',
+                        }}
+                      >
```

Comment:
A pain point w/ React is that passing an inline object like this will re-create the object on every render cycle.

The component that receives the object prop does a shallow equality check to see if its incoming data has changed. So in this case, on every render cycle, `<TagLabel />` will _think_ it's getting a different set of props and re-render itself. Even tho the dict is the same

In this specific case, it's not a big problem. But if `<TagLabel/>` was a heavy component, we would be wasting a lot of CPU cycles.

We can use inline style props to get around this like you've done elsewhere

```suggestion
                      <TagLabel maxWidth="50px" overflow="hidden" textOverflow="ellipsis" whiteSpace="nowrap">
```

It's unfortunate that we need to worry about this stuff but that's how React works right now.